containers/src/Kokkos_DualView.hpp

```diff
@@ -928,11 +928,26 @@ class DualView : public ViewTraits<DataType, Properties...> {
 
     if (sizeMismatch) {
       ::Kokkos::realloc(arg_prop, d_view, n0, n1, n2, n3, n4, n5, n6, n7);
-      if constexpr (alloc_prop_input::initialize) {
-        h_view = create_mirror_view(typename t_host::memory_space(), d_view);
+      if constexpr (alloc_prop_input::sequential_host_init) {
+        if constexpr (alloc_prop_input::initialize) {
+          h_view = create_mirror_view(
+              Kokkos::view_alloc(typename t_host::memory_space(),
+                                 Kokkos::SequentialHostInit),
+              d_view);
+        } else {
+          h_view = create_mirror_view(
+              Kokkos::view_alloc(Kokkos::WithoutInitializing,
+                                 typename t_host::memory_space(),
+                                 Kokkos::SequentialHostInit),
```

Comment:
There is a `static_assert` in the constructor of `ViewCtorProp` that seems to prevent passing both `WithoutInitializing` and `SequentialHostInit`:

- https://github.com/kokkos/kokkos/blob/3fe17d96d42eb2bd84f1f2a835a2ed0ba87606b6/core/src/View/Kokkos_ViewCtor.hpp#L208-L210